src/shared/seccomp-util.c

```diff
@@ -2019,29 +2019,6 @@ int seccomp_restrict_archs(Set *archs) {
         return 0;
 }
 
-int parse_syscall_archs(char **l, Set **ret_archs) {
-        _cleanup_set_free_ Set *archs = NULL;
-        int r;
-
-        assert(l);
-        assert(ret_archs);
-
-        STRV_FOREACH(s, l) {
-                uint32_t a;
-
-                r = seccomp_arch_from_string(*s, &a);
-                if (r < 0)
-                        return -EINVAL;
-
-                r = set_ensure_put(&archs, NULL, UINT32_TO_PTR(a + 1));
-                if (r < 0)
-                        return -ENOMEM;
-        }
-
-        *ret_archs = TAKE_PTR(archs);
-        return 0;
-}
```

Comment:
maybe instead of dropping this, just make the code that parses the SystemCallArchitectures property use this. It was clearly intended for that (added in same PR. Should be a trivial patch to bus_exec_context_set_transient_property